.github/workflows/test-and-push.yaml

```diff
@@ -29,56 +35,25 @@ jobs:
           pip install yamllint
           yamllint .
 
-  unit-tests:
-    runs-on: ubuntu-latest
-    permissions:
-      contents: write
-      pull-requests: write
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0 # Required for branch detection
-          ref: ${{ github.head_ref || github.ref_name }} # Handle PR branches
+      # --- Build Docker Image ---
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
```

Comment:
Is this actually needed? From what I can tell, it's to setup docker [buildx](https://github.com/docker/buildx) to be used. But I don't see any mention of buildx in the codebase.

If your intent was that it was needed for `docker build`, that is not the case as docker comes with `ubuntu-latest` which is what this workflow runs on.